byzcoin/service.go

```diff
@@ -1127,8 +1127,9 @@ func (s *Service) updateTrieCallback(sbID skipchain.SkipBlockID) error {
 	if nodeInNew {
 		// Update or start heartbeats
 		if s.heartbeats.exists(string(sb.SkipChainID())) {
-			log.Lvlf2("%s updating heartbeat monitor for %x with window %v", s.ServerIdentity(), sb.SkipChainID(), interval*rotationWindow)
-			s.heartbeats.updateTimeout(string(sb.SkipChainID()), interval*rotationWindow)
+			if s.heartbeats.updateTimeout(string(sb.SkipChainID()), interval*rotationWindow) {
```

Comment:
this can't be conditional, or else it creates a data race (see above). Instead, make the message something about "sending a heartbeat update to for %v", I guess.

Reply:
OK, but then I have to put the debugging-level to 3, else it spams too much...